internal/resolution/datasource/maven_registry.go

```diff
@@ -24,31 +24,95 @@ func NewMavenRegistryAPIClient(registry string) *MavenRegistryAPIClient {
 
 var errAPIFailed = errors.New("API query failed")
 
+// GetProject fetches a pom.xml specified by groupID, artifactID and version and parses it to maven.Project.
+// For a snapshot version, version level metadata is used to find the extact version value.
 func (m *MavenRegistryAPIClient) GetProject(ctx context.Context, groupID, artifactID, version string) (maven.Project, error) {
-	u, err := url.JoinPath(m.registry, strings.ReplaceAll(groupID, ".", "/"), artifactID, version, fmt.Sprintf("%s-%s.pom", artifactID, version))
+	if !strings.HasSuffix(version, "SNAPSHOT") {
+		return m.getProject(ctx, groupID, artifactID, version, "")
+	}
+
+	// Fetch version metadata for snapshot versions.
+	metadata, err := m.getVersionMetadata(ctx, groupID, artifactID, version)
+	if err != nil {
+		return maven.Project{}, err
+	}
+
+	snapshot := ""
+	for _, sv := range metadata.Versioning.SnapshotVersions {
+		if sv.Extension == "pom" {
+			// We only look for pom.xml for project metadata.
```

Comment:
Do you know what happens if the `pom` and `jar` versions are mismatched?

e.g. `foo-1.0.0-SNAPSHOT` has `jar` = `1.0.0-1` and `pom` = `1.0.0-2`.
Presumably, maven would download `foo-1.0.0-1.jar` for my dependency, but would it look for `foo-1.0.0-1.pom` or `foo-1.0.0-2.pom` for its dependencies?

Reply:
I don't know exact answer to this question. A snapshot version is the version in development before release and the actual version string is the timestamp + build number - I would expect them to be the same for the same snapshot version.